lib/MeshBVH.js

```diff
@@ -338,8 +99,8 @@ class MeshBVH extends MeshBVHNode {
 
 		if ( ! geo.boundingBox ) geo.computeBoundingBox();
 
-		const n = createNode( origTris, ( new THREE.Box3() ).copy( geo.boundingBox ), this );
-		while ( queue.length ) queue.pop()();
+		const n = createNode( boundsToArray( geo.boundingBox ), this, 0, ctx.tris.length );
+		n.index = new THREE.BufferAttribute( indices, 1 );
```

Comment:
Why not just take over the existing index array or generate one if it doesn't exist and use that?

Reply:
Well, the way we construct the new index array is by incrementally filling it up from left to right as we bottom out at leaf nodes. If we modified the old array instead, we might be overwriting indices that we still need to read from for future tree construction (because the old array isn't in sorted-triangle-order -- see the above explanation of how we're using `tris` to keep track of the sort order instead of modifiying the index directly.)

It's a good point that we wouldn't have to allocate a new index if we swapped the index data instead of using `tris`.